### ISSUE TITLE
Mismatch between torch.stft and torch.rfft

### ISSUE THREAD

**Issue by XPBooster** [pytorch/pytorch#85266]
### 🐛 Describe the bug

We have the test as follow
```
import torch
import matplotlib.pyplot as plt
t = [i for i in range(100)]
t = torch.tensor(t)
y = torch.sin(t)
output1 = torch.fft.rfft(y)
output2 = torch.stft(y, n_fft=100, hop_length=101, onesided=True, return_complex=True).squeeze()
plt.plot(output1)
plt.plot(output2)
plt.show()
```
where n_hop=101 to ensure there is only 1 round of fft. As such, we expect to have **output1=output2**, since all settings are kept the same, i.e., onesided, normalize=None. However, that does not happen.
![image](https://user-images.githubusercontent.com/58415678/191063389-41e731e4-fd7d-4cf0-82ec-5c9b91d3bbfc.png)

Why there is a **large** mismatch between output1 and output2?

### Versions

pytorch 1.12

cc @mruberry @peterbell10 @kshitij12345

**Comment by peterbell10** [comment]
> Why there is a large mismatch between output1 and output2?

By default, `stft` centers the input frame by padding the input on the left and right by `n_fft // 2`. If we run the same experiment with `center=False`, there is no discrepancy. 

Alternatively, to match the centered stft call, we can pad the input to rfft: 
```python
rfft_input = torch.nn.functional.pad(y.view(1, -1), [50, 50], mode='reflect').squeeze()[:100]
output3 = torch.fft.rfft(rfft_input)
assert torch.allclose(output2, output3)
```
